src/set_target_node.cpp

```diff
@@ -0,0 +1,112 @@
+#include <rt_assignment_2/set_target_node.hpp>
+#include <rt_assignment_2/RobotState.h>
+#include <rt_assignment_2/RobotTarget.h>
+
+#include <ros/ros.h>
+#include <ros/timer.h>
+
+#include <boost/bind.hpp>
+
+#include <actionlib/client/simple_action_client.h>
+#include <nav_msgs/Odometry.h>
+
+#include <assignment_2_2023/PlanningAction.h>
+
+
+
+namespace rt_assignment_2
+{
+
+    SetTargetNode::SetTargetNode(
+        const std::shared_ptr<actionlib::SimpleActionClient<assignment_2_2023::PlanningAction>> &ac,
+        const std::shared_ptr<ros::NodeHandle> &nh)
+        : ac_(std::move(ac)), nh_(std::move(nh))
+    
+    {
+        ROS_INFO("Waiting for server to connect");
+        ac->waitForServer(ros::Duration(2.0));
+
+        // Subscribers
+        sub_target_ = nh_->subscribe("/robot_target", 10, &SetTargetNode::setNewTargetCallback, this);
+        sub_state_ = nh_->subscribe("/odom", 10, &SetTargetNode::robotStateCallback, this);
+        sub_cancel_ = nh_->subscribe("/robot_cancel_goal", 10, &SetTargetNode::cancelGoalCallback, this);
+
+        // Publishers
+        pub_state_ = nh_->advertise<rt_assignment_2::RobotState>("/robot_state", 1000);
+
+        // Timers
```

Comment:
Adding the frequency of the timer would be nice in here. I will be simpler than calculating the `1 / 0.5`